crates/db_views_actor/src/community_moderator_view.rs

```diff
@@ -57,23 +59,33 @@ impl CommunityModeratorView {
       .await
   }
 
-  pub async fn for_person(
+  pub async fn for_person<'a>(
     pool: &mut DbPool<'_>,
     person_id: PersonId,
-    is_authenticated: bool,
+    local_user: Option<&'a LocalUser>,
   ) -> Result<Vec<Self>, Error> {
     let conn = &mut get_conn(pool).await?;
     let mut query = community_moderator::table
       .inner_join(community::table)
       .inner_join(person::table)
       .filter(community_moderator::person_id.eq(person_id))
-      .filter(community::deleted.eq(false))
-      .filter(community::removed.eq(false))
       .select((community::all_columns, person::all_columns))
       .into_boxed();
-    if !is_authenticated {
+
+    if local_user.is_none() {
```

Comment:
Use the LocalUserOptionHelper::visible_communities_only method

Reply:
Good call.